docs/guides/disagg_perf_tuning.md

```diff
@@ -17,82 +17,127 @@ limitations under the License.
+## Engine Tuning
 
-Number of GPUs | Performance
---- | ---
-Cannot hold weights in VRAM | OOM
-(Barely hold weights in VRAM) | (KV cache is too small to maintain large enough sequence length or reasonable batch size)
-Minimum number with fair amount of KV cache | Best overall throughput/GPU, worst latency/user
-Between minimum and maximum | Tradeoff between throughput/GPU and latency/user
-Maximum number limited by communication scalability | Worst overall throughput/GPU, best latency/user
-More than maximum | Communication overhead dominates, poor performance
+The most important engine knob to tune is the parallelization mapping.
+For most dense models, the best setting is to use TP within node and PP across nodes.
+For example, for llama 405b w8a8 on H100, TP8 on a single node or TP8PP2 on two nodes is usually the best choice.
+The next thing to decide is how many numbers of GPU to serve the model.
+Typically, the number of GPUs vs the performance follows the following pattern:
 
-Note that for decode-only engines, sometimes larger number of GPUs has to larger kv cache per GPU and more decoding requests running in parallel, which leads to both better throughput/GPU and better latency/user. For example, for llama3.3 70b NVFP4 quantization on B200 in vllm with 0.9 free GPU memory fraction:
+| Number of GPUs                                      | Performance
+| :-------------------------------------------------- | :---------------------------------------------------------------------------------------- |
+| Cannot hold weights in VRAM                         | OOM                                                                                       |
+| (Barely hold weights in VRAM)                       | (KV cache is too small to maintain large enough sequence length or reasonable batch size) |
+| Minimum number with fair amount of KV cache         | Best overall throughput/GPU, worst latency/user                                           |
+| Between minimum and maximum                         | Tradeoff between throughput/GPU and latency/user                                          |
+| Maximum number limited by communication scalability | Worst overall throughput/GPU, best latency/user                                           |
+| More than maximum                                   | Communication overhead dominates, poor performance                                        |
 
-TP Size | KV Cache Size (GB) | KV Cache per GPU (GB) | Per GPU Improvement over TP1
---- | --- | --- | ---
-1 | 113 | 113 | 1.00x
-2 | 269 | 135 | 1.19x
-4 | 578 | 144 | 1.28x
+Note that for decode-only engines, sometimes larger number of GPUs has to larger KV cache per GPU and more decoding requests running in parallel, which leads to both better throughput/GPU and better latency/user.
```

Comment:
I am not sure what this sentence is saying

Reply:
neither do I, again not my prose. I was just reformatting for readability.